pkg/optimization/substat_details.go

```diff
@@ -0,0 +1,551 @@
+package optimization
+
+import (
+	"fmt"
+	"github.com/genshinsim/gcsim/internal/simulator"
+	"github.com/genshinsim/gcsim/pkg/core"
+	"math"
+	"sort"
+	"strings"
+)
+
+type SubstatOptimizerDetails struct {
```

Comment:
design wise i think this structure feels a bit icky. at a cursory glance i feel like the structure should be more like soemthing like the following

```
type Optimizer struct {
   Input SomeStruct
   WIP SomeStruct
}
```

where `Input` is your starting point and contains all the relevant information like the config, initial profile etc... and `WIP` (could be a slice here) is keeping track of each iteration where you try different stats to get to the optimized state

this current design/data structure feels a bit like a big dumb of different variables not organized in any fashion